### PR TITLE
Add HEP process readme

### PR DESCRIPTION
**Pull request by bk201**
Add a readme to describe the process of HEP.

### CODE REVIEW

File: enhancements/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Harvester Enhancement Proposal (HEP)
+
+This directory collects Harvester Enhancement Proposals (HEP). HEPs are used to describe highlighted features, enhancements, or bug fixes that require significant changes to the code or architecture. You can use the [template](./YYYYMMDD-template.md) to create a HEP pull request.
```
Comment (guangbochen) [inline enhancements/README.md]:
```suggestion
This directory serves as a collection of Harvester Enhancement Proposals (HEPs). HEPs are used to describe highlighted features, enhancements, or bug fixes that require substantial changes to the code or architecture. To create a HEP, you can use the provided [template](./YYYYMMDD-template.md) and submit a pull request.
```

File: enhancements/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+The following steps describe how HEPs are involved in the development process:
+
+1. An Engineer or contributor gets assigned a Github issue. The GitHub issue should have the `require/HEP` label.
+1. When the assignee decides to work on it (which likely means it’s now targeting the next minor release), they will start working on the HEP if HEP is required. It might take a while because it also involves POC and lots of discussions.
+1. Once the assignee has an initial design, he or she will submit a PR containing the HEP to the `harvester/harvester` GitHub repo, `enhancements` directory. For example https://github.com/harvester/harvester/pull/3673.
+1. More discussion will start on the HEP, now it’s time we can get feedback from the community or team.
+1. HEP might be not getting merged for a while. The assignee might decide to implement the feature more and come back on updating the HEP. It’s generally required HEP to be merged when the feature is merged, so the HEP is at least going to reflect how the first iteration of works.
```
Comment (guangbochen) [inline enhancements/README.md]:
```suggestion
1. An engineer or contributor is assigned a GitHub issue. The GitHub issue should have the `require/HEP` label.
1. Once the assignee decides to work on the issue (typically targeting the next minor release), they will begin working on the HEP first. This process may involve proof of concept and extensive discussions.
1. After the assignee has a preliminary design, they will submit a pull request containing the HEP to the `harvester/harvester` GitHub repository's `enhancements` directory. For example, you can find a HEP pull request at https://github.com/harvester/harvester/pull/3673.
1. The HEP will undergo further discussion, allowing for feedback from the community or team.
1. The HEP may not be merged immediately. The assignee may choose to implement the feature further before updating the HEP. Generally, it is required to merge the HEP when the feature itself is merged, and the HEP will ensure that at least it reflects the initial iteration of work."
```

File: enhancements/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+The following steps describe how HEPs are involved in the development process:
+
+1. An engineer or contributor is assigned a GitHub issue. The GitHub issue should have the `require/HEP` label.
+1. Once the assignee decides to work on the issue (typically targeting the next minor release), they will begin working on the HEP first. This process may involve proof of concept and extensive discussions.
+1. After the assignee has a preliminary design, they will submit a pull request containing the HEP to the `harvester/harvester` GitHub repository's `enhancements` directory. For example, you can find a HEP pull request at https://github.com/harvester/harvester/pull/3673.
+1. The HEP will undergo further discussion, allowing for feedback from the community or team.
+1. The HEP may not be merged immediately. The assignee may choose to implement the feature further before updating the HEP. Generally, it is required to merge the HEP when the feature itself is merged, and the HEP will ensure that at least it reflects the initial iteration of work."
```
Comment (yasker) [inline enhancements/README.md]: Extra `"` at the end of the sentence.

File: enhancements/README.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Harvester Enhancement Proposal (HEP)
+
+This directory serves as a collection of Harvester Enhancement Proposals (HEPs). HEPs are used to describe highlighted features, enhancements, or bug fixes that require substantial changes to the code or architecture. To create a HEP, you can use the provided [template](./YYYYMMDD-template.md) and submit a pull request.
+
+The following steps describe how HEPs are involved in the development process:
+
+1. An engineer or contributor is assigned a GitHub issue. The GitHub issue should have the `require/HEP` label.
```
Comment (yasker) [inline enhancements/README.md]: Can add 
`If the issue doesn't have require/HEP initially but after the investigation, the assigned engineer or contributor thinks it's preferred to have a HEP, they can add the label or contact the project's maintainer to add the label.`

**Review by yasker — CHANGES_REQUESTED** [review]

File: enhancements/README.md (hunk 4 above)
Comment (bk201) [reply enhancements/README.md]: Added, thanks!

File: enhancements/README.md (hunk 3 above)
Comment (bk201) [reply enhancements/README.md]: Fixed!

**Review by yasker — APPROVED** [review]

File: enhancements/README.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# Harvester Enhancement Proposal (HEP)
+
+This directory serves as a collection of Harvester Enhancement Proposals (HEPs). HEPs are used to describe highlighted features, enhancements, or bug fixes that require substantial changes to the code or architecture. To create a HEP, you can use the provided [template](./YYYYMMDD-template.md) and submit a pull request.
+
+The following steps describe how HEPs are involved in the development process:
+
+1. An engineer or contributor is assigned a GitHub issue. The GitHub issue should have the `require/HEP` label. If the issue doesn't have require/HEP initially but after the investigation, the assigned engineer or contributor thinks it's preferred to have a HEP, they can add the label or contact the project's maintainer to add the label.
```
Comment (LucasSaintarbor) [inline enhancements/README.md]:
```suggestion
1. An engineer or contributor is assigned a GitHub issue. The GitHub issue should have the `require/HEP` label. If the issue doesn't have the `require/HEP` label initially but after investigation, the assigned engineer or contributor thinks the issue should have a HEP, they can add the label or contact the project's maintainer to add the label.
```

File: enhancements/README.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+1. Once the assignee decides to work on the issue (typically targeting the next minor release), they will begin working on the HEP first. This process may involve proof of concept and extensive discussions.
+1. After the assignee has a preliminary design, they will submit a pull request containing the HEP to the `harvester/harvester` GitHub repository's `enhancements` directory. For example, you can find a HEP pull request at https://github.com/harvester/harvester/pull/3673.
+1. The HEP will undergo further discussion, allowing for feedback from the community or team.
+1. The HEP may not be merged immediately. The assignee may choose to implement the feature further before updating the HEP. Generally, it is required to merge the HEP when the feature itself is merged, and the HEP will ensure that at least it reflects the initial iteration of work.
```
Comment (LucasSaintarbor) [inline enhancements/README.md]:
```suggestion
1. The HEP might not be merged immediately. The assignee may choose to implement the feature further before updating the HEP. Generally, it is required to merge the HEP when the feature itself is merged. The HEP will ensure that the feature work reflects the initial iteration of work.
```

**Review by LucasSaintarbor — APPROVED** [review]
Made some small changes. LGTM 👍

File: enhancements/README.md (hunk 6 above)
Comment (bk201) [reply enhancements/README.md]: Applied; thanks for the reviewing.

**Review by guangbochen — APPROVED** [review]

**Pull request merged**